Code/Framework/AzCore/AzCore/Debug/PerformanceCollector.cpp

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/IO/OpenMode.h>
+#include <AzCore/IO/GenericStreams.h>
+#include <AzCore/IO/ByteContainerStream.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Date/DateFormat.h>
+
+#include "PerformanceCollector.h"
+
+
+namespace AZ::Debug
+{
+    PerformanceCollector::PerformanceCollector(const AZStd::string_view logCategory, AZStd::span<const AZStd::string_view> m_metricNames, OnBatchCompleteCallback onBatchCompleteCallback)
+        : m_logCategory(logCategory), m_onBatchCompleteCallback(onBatchCompleteCallback)
+    {
+        for (const auto& metricName : m_metricNames)
+        {
+            [[maybe_unused]] const auto statisticPtr = m_statisticsManager.AddStatistic(metricName, metricName, "us");
+            AZ_Assert(statisticPtr, "Failed to add metric with name <%.*s>. Maybe already added?", AZ_STRING_ARG(metricName));
+        }
+        RestartPeriodicEventStamps();
+    }
+
+    PerformanceCollector::~PerformanceCollector()
+    {
+
+    }
+
+    bool PerformanceCollector::IsWaitingBeforeCapture()
+    {
+        return !m_numberOfCaptureBatches || m_isWaitingBeforeNextBatch;
+    }
+
+    void PerformanceCollector::FrameTick()
+    {
+        if (!m_numberOfCaptureBatches)
+        {
+            return;
+        }
+
+        if (m_isWaitingBeforeNextBatch)
+        {
+            if (m_waitTimeBeforeEachBatch.count() > 0)
+            {
+                AZStd::chrono::steady_clock::time_point now = AZStd::chrono::steady_clock::now();
+                if (m_startWaitTime.time_since_epoch().count() == 0)
+                {
+                    AZ_TracePrintf(LogName, "Will Wait %lld seconds before starting batch %u...", m_waitTimeBeforeEachBatch.count(), m_numberOfCaptureBatches);
```

Comment:
You may want to get rid of these TracePrintfs or put them in an `if constexpr (SomeConstantThatIsFalse)`.

I know noticed capturing benchmarks using the EventLogger, the `AZ_TracePrintf` statement took a significant amount of time on the order of mircroseconds.

Also nitpick, they need a newline character added as well.

Reply:
These are very low frequency print statements.